src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -487,54 +141,13 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
             }
         }
 
-        internal sealed override bool HasSpecialName
-        {
-            get
-            {
-                if (this.HasRuntimeSpecialName)
-                {
-                    return true;
-                }
-
-                var data = GetDecodedWellKnownAttributeData();
-                return data != null && data.HasSpecialNameAttribute;
-            }
-        }
-
         internal sealed override bool HasRuntimeSpecialName
         {
             get
             {
                 return this.Name == WellKnownMemberNames.EnumBackingFieldName;
```

Comment:
>return this.Name == WellKnownMemberNames.EnumBackingFieldName; [](start = 16, length = 62)

It feels like we should at least check that the containing type is an enum. Please open an issue to follow-up

Reply:
Filed [https://github.com/dotnet/roslyn/issues/23058](https://github.com/dotnet/roslyn/issues/23058)

---
In reply to: [149507555](https://github.com/dotnet/roslyn/pull/22664#discussion_r149507555) [](ancestors = 149507555)